src/scripts/worldmap/MapHelper.ts

```diff
@@ -210,52 +208,51 @@ class MapHelper {
         if (TownList[townName]?.ignoreAreaStatus) {
             return '';
         }
+        const states = new Set([areaStatus.completed]);
         // Check if this location is locked
         if (!MapHelper.accessToTown(townName)) {
-            return areaStatus[areaStatus.locked];
+            states.add(areaStatus.locked);
         }
-        const states = [];
         // Is this location a dungeon
         if (dungeonList[townName] && dungeonList[townName].isUnlocked()) {
             const possiblePokemon = dungeonList[townName].allAvailablePokemon();
             const shadowPokemon = dungeonList[townName].allAvailableShadowPokemon();
 
             if (!App.game.statistics.dungeonsCleared[GameConstants.getDungeonIndex(townName)]()) {
-                states.push(areaStatus.incomplete);
+                states.add(areaStatus.incomplete);
             } else if (dungeonList[townName].isThereQuestAtLocation()) {
```

Comment:
Don't these also need to be split into individual `if` blocks so all the states can get added?

Reply:
... maybe

Reply:
Did this and the TownContent classes too